coderedcms/models/page_models.py

```diff
@@ -766,21 +766,28 @@ def upcoming_occurrences(self):
         """
         return self.query_occurrences(num_of_instances_to_return=10)
 
+
+    def _most_recent_occurrence(self):
+        aoc = []
+        for occurrence in self.occurrences.all():
+            aoc += [instance for instance in occurrence.all_occurrences()]
+        if len(aoc) > 0:
+            return aoc[-1] # last one in the list
+
     @property
-    def most_recent_occurrence(self):
+    def next_or_most_recent_occurrence(self):
         """
         Gets the next upcoming, or last occurrence if the event has no more occurrences.
         """
-        noc = self.next_occurrence()
-        if noc:
-            return noc
-        else:
-            aoc = []
-            for occurrence in self.occurrences.all():
-                aoc += [instance for instance in occurrence.all_occurrences()]
-            if len(aoc) > 0:
-                return aoc[-1] # last one in the list
-            return None
+        try:
+            noc = self.next_occurrence()
+            return noc if noc else self._most_recent_occurrence()
+        except AttributeError:
+            # Triggers when a preview is initiated on an EventPage because it uses a FakeQuerySet object.
+            # Here we manually compute the next_occurrence
+            occurrences = [e.next_occurrence() for e in self.occurrences.all()]
```

Comment:
So the issue was that `self.next_occurrence()` breaks on preview, but `self.occurrences.all()[0].next_occurrence()` DOES work? I guess that sorta makes sense if there is a related_name that does not exist with the FakeQuerySet. I would be fine just showing the first occurrence on a preview, which would simplify and reduce the resources needed to generate a preview of this page.